provider_test.go

```diff
@@ -0,0 +1,81 @@
+package testcontainers
+
+import (
+	"os"
+	"testing"
+)
+
+// TestProviderTypeGetProviderAutodetect should NOT be run in parallel as it sets environment variables.
```

Comment:
Using `t.Setenv` would resolve that limitation

Reply:
Thanks! I didn't know about `t.Setenv()`. Nice to learn new tricks.

According to the docs, it still can't be used in Parallel tests, but it is a lot simpler. I will switch to using it.